File: netCore/NEC.UnitTest/src/NEC.Models/Seat.cs
using System;

namespace NEC.Models
{
    public class Seat
    {
        private readonly SeatBelt _seatBelt;

        public Seat(SeatBelt seatBelt) => 
            _seatBelt = seatBelt ?? throw new ArgumentNullException(nameof(seatBelt));

        public bool IsSeatBeltFastened() => _seatBelt.IsFastened();

        public void FastenSeatBelt() => _seatBelt.Fasten();
    }
}

File: java/nec.unit-test/src/main/java/nec/Models/Seat.java
package nec.Models;

public class Seat {

    private SeatBelt _seatBelt;

    public Seat(SeatBelt seatBelt) {

	if (seatBelt == null) {
	    throw new IllegalArgumentException();
	}

	this._seatBelt = seatBelt;
    }

    public Boolean isSeatBeltFastened() {

	return this._seatBelt.isFastened();
    }

    public void fastenSeatBelt() {

	this._seatBelt.fasten();
    }
}
